pages/extensions/extract_highlights_info/page_info_collector.py

```diff
@@ -1,9 +1,17 @@
 # -*- coding: utf-8 -*-
 import os
+import re
 import json
 from grow.pods.pods import Pod
 from grow.documents.document import Document
 
+FIRST_SENTENCE_PATTERN = re.compile(
+    # We also match HTML comments and markup [tag]...[/tag] blocks to skip them
+    r'<!--.*?-->|^\[(\w+)(?:[:\s][^\]]+)\].*?\[/\1\]|(^\w[^.\n]*(?:\n\w[^.\n]*)*(?:\.|$))',
+    re.MULTILINE | re.DOTALL)
```

Comment:
This seems a bit fragile:

* What about other markdown features such as : `**Bold** _italic_ ...`? 
* What if a sentence ends with `!` or `?`? 
* What about other languages?

Reply:
Very good points!

I changed the logic, so that we first create the html for the markup, remove tags and blocks we do not want and then search the first block.

The first sentence can now end with . or ! or ?

Also \w will match any unicode character, so that also arabic sentences are found.